src/types.jl

```diff
@@ -94,6 +100,15 @@ struct TimestampedAnnotationList
     onset_in_seconds::Float64
     duration_in_seconds::Union{Float64,Nothing}
     annotations::Vector{String}
+    function TimestampedAnnotationList(onset_in_seconds, duration_in_seconds, annotations)
+        onset_in_seconds = _nearest_representable_edf_time_value(onset_in_seconds)
+        duration_in_seconds = _nearest_representable_edf_time_value(duration_in_seconds)
+        return new(onset_in_seconds, duration_in_seconds, annotations)
+    end
+end
+
+function _nearest_representable_edf_time_value(x)
+    return round(x, digits=(8 - (ndigits(floor(Int, x)) + signbit(x) + isinteger(x))))
```

Comment:
```suggestion
    return round(x; digits=(8 - (ndigits(floor(Int, x)) + signbit(x) + isinteger(x))))
```
Not every day you get to do a style suggestion for the person who wrote the style guide :stuck_out_tongue:

Reply:
CI is failing because `floor` is getting a `nothing` somewhere :thinking:

Reply:
Ah, I see, `duration_in_seconds` is allowed to be `nothing`. So you could add a method for `::Nothing` that just returns `nothing`.